modules/core/map/js/farmOS.map.behaviors.input.js

```diff
@@ -1,11 +1,56 @@
 (function () {
   farmOS.map.behaviors.input = {
     attach: function (instance) {
+
+      // Get the data input form element.
+      var input = instance.map.getTargetElement().parentElement.querySelector('[data-map-geometry-field]')
+
+      // When features change in the edit layer, write WKT to the input form element.
       instance.editAttached && instance.editAttached.then(() => {
         instance.edit.wktOn('featurechange', function(wkt) {
-          instance.map.getTargetElement().parentElement.querySelector('[data-map-geometry-field]').value = wkt;
+          input.value = wkt;
         });
       });
+
+      // Run a handleInput() callback when input changes and attempt to
+      // import the new WKT.
+      input.oninput = handleInput;
+      function handleInput(e) {
+        if (input.value) {
+
+          // Add a new temporary invisible WKT layer.
+          var layer = instance.addLayer('wkt', {
+            title: 'WKT',
+            wkt: input.value,
+            visible: false,
+          });
+
+          // Get the new layer's features. Bail if there are none.
+          var features = layer.getSource().getFeatures()
+
+          // If there are no features, bail.
+          if (!features) {
```

Comment:
I believe `ol/source/Vector::getFeatures()` always returns an array - even if the source is empty... https://github.com/openlayers/openlayers/blob/9613cb97617ce2e3af082c236d5f63a9dc7e93f3/src/ol/source/Vector.js#L700-L712

```suggestion
          if (!features || !features.length) {
```

Reply:
I actually removed this condition and forgot to force push my branch. It had a logical error which could have led to the code returning before cleaning up the temporary layer. This also changes slightly when we are able to use this: https://github.com/farmOS/farmOS-map/pull/194

So it's worth reviewing again with these changes. I just rebased with this change, plus some other fixup commits to address your other feedback.